src/main/java/com/aws/greengrass/clientdevices/auth/certificate/ClientCertificateGenerator.java

```diff
@@ -60,12 +60,17 @@ public synchronized void generateCertificate(Supplier<List<String>> connectivity
             return;
         }
 
+        if (certificateStore.getCaPrivateKey() == null) {
```

Comment:
When would this be null?

Reply:
As you commented out earlier, this should not be null if the component is in `RUNNING` state. But because we're using `serviceErrored(e)` to put the component in BROKEN state, it takes some amount of time for the Nucleus to identify that this component is BROKEN  and retry. But, in this time, CDA continues to execute next steps and sometimes resulting in null pointer exceptions.


I've seen this happening here while trying to get certificates after the service errored https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/ff646044c87875f465dba5cccd0b8f5b54132ad1/src/main/java/com/aws/greengrass/clientdevices/auth/ClientDevicesAuthService.java#L272-L276 but I added it for both cert  and key.